src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -820,14 +820,16 @@ private void BuildCookieCollectionFromDomainMatches(Uri uri, bool isSecure, int
 
                 lock (pathList.SyncRoot)
                 {
-                    foreach (DictionaryEntry entry in pathList)
```

Comment:
Ah, I misread the code.  I thought we were switching from using a dictionary's struct enumerator, in which case there wouldn't have been any boxing.  Makes sense.